drivers/digital-io/max22196/max22196.c

```diff
@@ -0,0 +1,528 @@
+ * @param delay - Filter delay value to be read for the MAX22196's specified
+ * 		  channel.
+ * @return 0 in case of succes, negative error code otherwise.
+*/
+int max22196_filter_get(struct max22196_desc *desc, uint32_t ch,
+			uint32_t *flt_en, enum max22196_delay *delay)
+{
+	int ret;
+	uint32_t del_val;
+
+	if (ch > MAX22196_CHANNELS - 1)
+		return -EINVAL;
+
+	ret = max22196_reg_read(desc, MAX22196_CFG_REG(ch), &del_val);
+	if (ret)
+		return ret;
+
+	*delay = no_os_field_get(MAX22196_DELAY_MASK, del_val);
+	*flt_en = no_os_field_get(MAX22196_FLTEN_MASK, del_val);
+
+	return 0;
+}
+
+/**
+ * @brief Set fault mask bits in the fault mask registers.
+ * @param desc - MAX22196 device descriptor.
+ * @param fault_mask - Requested fault to set mask to.
+ * @param enabled - Mask state to be set.
+ * @return - 0 in case of succes, negative error code otherwise.
+*/
+int max22196_fault_mask_set(struct max22196_desc *desc,
+			    enum max22196_fault_mask fault_mask,
+			    bool enabled)
+{
+	switch (fault_mask) {
+	case MAX22196_FAULT1_VMLOW ... MAX22196_FAULT1_FAULT2:
+		return max22196_reg_update(desc, MAX22196_F1MASK_REG,
+					   MAX22196_FAULT_MASK(fault_mask),
+					   no_os_field_prep(MAX22196_FAULT_MASK(fault_mask),
+							   enabled));
+	case MAX22196_FAULT2_RFDIS ... MAX22196_FAULT2_VAUV:
+		return max22196_reg_update(desc, MAX22196_F1MASK_REG,
+					   MAX22196_FAULT_MASK(fault_mask - MAX22196_FAULT2_RFDIS),
+					   no_os_field_prep(MAX22196_FAULT_MASK(fault_mask - MAX22196_FAULT2_RFDIS),
+							   enabled));
+	default:
+		return -EINVAL;
+	}
+}
+
+
+/**
+ * @brief Get fault mask bits from the fault mask registers.
+ * @param desc - MAX22196 device descriptor.
+ * @param fault_mask - Requested fault to get mask from.
+ * @param enabled - Mask state to be read.
+ * @return - 0 in case of succes, negative error code otherwise.
+*/
+int max22196_fault_mask_get(struct max22196_desc *desc,
+			    enum max22196_fault_mask fault_mask,
+			    bool *enabled)
+{
+	int ret;
+	uint32_t reg_val;
+
+	switch (fault_mask) {
+	case MAX22196_FAULT1_VMLOW ... MAX22196_FAULT1_FAULT2:
+		ret =  max22196_reg_read(desc, MAX22196_F1MASK_REG, &reg_val);
+		if (ret)
+			return ret;
+		*enabled = no_os_field_get(MAX22196_FAULT_MASK(fault_mask),
+					   reg_val);
+		break;
+	case MAX22196_FAULT2_RFDIS ... MAX22196_FAULT2_VAUV:
+		ret =  max22196_reg_read(desc, MAX22196_F2MASK_REG, &reg_val);
+		if (ret)
+			return ret;
+		*enabled = no_os_field_get(MAX22196_FAULT_MASK(fault_mask -
+					   MAX22196_FAULT2_RFDIS),
+					   reg_val);
+		break;
+	default:
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+/**
+ * @brief Set global configuration.
+ * @param desc - MAX22196 device descriptor.
+ * @param global_cfg - Specifc configuration to be set.
+ * @param enabled - State of the specific configuration.
+ * @return 0 in case of succes, negative error code otherwise.
+*/
+int max22196_global_cfg(struct max22196_desc *desc,
```

Comment:
in max22190 REFDISHTCFG is set in fault_set... I think we should have the same handling in both drivers. same with clrfiltr